src/Compilers/Core/Portable/Emit/EmitOptions.cs

```diff
@@ -177,6 +178,14 @@ public bool Equals(EmitOptions other)
                 return false;
             }
 
+            for (int i = 0; i < this.InstrumentationKinds.Length; i++)
+            {
+                if (this.InstrumentationKinds[i] != other.InstrumentationKinds[i])
+                {
+                    return false;
```

Comment:
The code looks different now, because @drognanar changed it to use SequenceEqual, but equality of EmitOptions still implies identical order of instrumentation kinds. That isn't too significant now, in that only one instrumentation kind can be specified, but seems questionable. Tagging @gafter @AlekseyTs @cston for opinions.